commit d55b6f4f1b2bfab96ad57e0239b0b2b57785df77
Author: Jiri Slaby <jirislaby@gmail.com>
Date:   Wed Feb 29 01:19:18 2012 -0800

    Don't fail if netfilter isn't enabled in the kernel.
    
    Check for EPROTONOSUPPORT as well as EAFNOSUPPORT.
    
    Reviewed-By: Guy Harris <guy@alum.mit.edu>

diff --git a/pcap-netfilter-linux.c b/pcap-netfilter-linux.c
index 225e49f..f9c6bef 100644
--- a/pcap-netfilter-linux.c
+++ b/pcap-netfilter-linux.c
@@ -452,8 +452,8 @@ netfilter_platform_finddevs(pcap_if_t **alldevsp, char *err_str)
 	
 	sock = socket(AF_NETLINK, SOCK_RAW, NETLINK_NETFILTER);
 	if (sock < 0) {
-		/* if netlink is not supported this this is not fatal */
-		if (errno == EAFNOSUPPORT)
+		/* if netlink is not supported this is not fatal */
+		if (errno == EAFNOSUPPORT || errno == EPROTONOSUPPORT)
 			return 0;
 		snprintf(err_str, PCAP_ERRBUF_SIZE, "Can't open netlink socket %d:%s",
 			errno, pcap_strerror(errno));
